### ISSUE TITLE
Provide an api to use the renderer

### ISSUE THREAD

**Issue by p-lr** [maplibre/maplibre-native#2241]
For use cases when we don't need a full fledged map, but only to convert pdf tiles into raster images, an api would be great.

For example, given this pbf tile : https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_v2/VectorTileServer/tile/6/23/31.pbf

And this [style](https://cdn.arcgis.com/sharing/rest/content/items/b079fb8620f945cca0019db1cf611b45/resources/styles/root.json?f=json), get the corresponding raster image.

**Comment by louwers** [comment]
What kind of API do you mean?

**Comment by p-lr** [comment]
Ideally, a just a blocking call such as:
```
encodePNG(Renderer.render(pathToPbfFile, style))
```
I has a look at the renderer tool you mention. It involves the creation of a Map, then a PNG is made from out of the map rendering IIUC.
What I would like is only to do that from vector tiles, without the need to create a Map instance.

**Comment by louwers** [comment]
Correct, you need a Map instance. But in that example, only a single frame is rendered.

MapLibre Native is [not really optimized](https://github.com/maplibre/maplibre/discussions/334) for static rendering (although it is used for this a lot as well). And it doesn't have a first-class API for it.

Changes in this area are welcome, although significant changes are best done by making a [design proposal](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2022-09-02-design-proposal-template.md) first. In the case of the API you suggest, we already have a `Renderer` class with a different job, so we would need another name. Also since you can specify a local `.pbf` file as a source in the style, I don't think it would be good to have a separate parameter for that specifically.

**Comment by p-lr** [comment]
Thanks for the clarification. If the library could offer a first class api for static rendering (of map area or just tiles), that would be awesome.

**Comment by wipfli** [comment]
https://github.com/ConservationMetrics/mapgl-tile-renderer might be an option for you @p-lr

**Comment by wipfli** [comment]
And for on-the-fly rendering of pngs, you can use maptiler/tileserver-gl.

**Comment by p-lr** [comment]
@wipfli The goal is to do this client side, on mobile devices, not server-side.
